### PR TITLE
Migrate to LLVM15

### PR DESCRIPTION
**Pull request by frabert**

### CODE REVIEW

**Comment by Ninja3047** [comment]
diff is failing to build due to llvm version mismatch, should get resolved after this gets merged

File: tools/xref/StmtPrinter.cpp [hunk 1]
```diff
@@ -185,11 +186,72 @@ class StmtPrinter : public StmtVisitor<StmtPrinter> {
 /// with no newline after the }.
 void StmtPrinter::PrintRawCompoundStmt(CompoundStmt *Node) {
   OS << "{" << NL;
+  PrintFPPragmas(Node);
   for (auto *I : Node->body()) PrintStmt(I);
 
   Indent() << "}";
 }
 
+void StmtPrinter::PrintFPPragmas(CompoundStmt *S) {
```
Comment (surovic) [inline tools/xref/StmtPrinter.cpp:195]: I assume this is a straight copy-paste from clang, right?

File: tools/xref/StmtPrinter.cpp [hunk 2]
```diff
@@ -751,13 +813,25 @@ void StmtPrinter::VisitOMPParallelMasterDirective(
   PrintOMPExecutableDirective(Node);
 }
 
+void StmtPrinter::VisitOMPParallelMaskedDirective(
```
Comment (surovic) [inline tools/xref/StmtPrinter.cpp:816]: Do we need the OpenMP pragma printers for anything?

**Review by surovic — APPROVED** [review]
Just to be sure, the changes in various printers are straight copy-pastes from clang, right? Otherwise LGTM.

**Comment by frabert** [comment]
All copy-pastes. I want to get rid of the custom printer when we update to LLVM16, but that might require using PASTA, which requires a patched Clang

File: tools/xref/StmtPrinter.cpp (hunk 2 above)
Comment (frabert) [reply tools/xref/StmtPrinter.cpp:816]: Just for making Clang happy and making tracking changes across versions easier

**Pull request merged**